dbms/src/Functions/FunctionsJson.h

```diff
@@ -491,37 +506,99 @@ class FunctionsJsonArray : public IFunction
 
     void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
     {
-        auto nested_block = createBlockWithNestedColumns(block, arguments, result);
+        if (arguments.empty())
```

Comment:
Why add this?

Reply:
It is an optimization for `json_array()`.
In the previous implementation, `ColumnConst` wouldn't be generated.